tests/unit_tests/scheduler/test_scheduler.py

```diff
@@ -204,6 +205,44 @@ async def wait():
     assert timeouteventfound
 
 
+@pytest.mark.parametrize("max_running", [0, 1, 2, 10])
+async def test_max_running(max_running, mock_driver, storage, tmp_path):
+    runs: List[bool] = []
+
+    async def wait():
+        nonlocal runs
+        runs.append(True)
+        await asyncio.sleep(0.01)
+        runs.append(False)
+
+    ensemble_size = max_running * 3 if max_running > 0 else 10
```

Comment:
Does this mean we should have 3x more ensembles than max_running? Maybe have it as a variable instead of magic number

Reply:
We want more realizations than max_running, if not we are exposing the scheduler for any problem related to max_running. Can add a comment, but we basically just need `ensemble_size > max_running`